conans/model/requires.py

```diff
@@ -268,19 +268,8 @@ def transform_downstream(self, pkg_type, require, dep_pkg_type):
             # TODO: We could implement checks in case private is violated (e.g shared libs)
             return
 
-        if require.build:  # public!
-            # TODO: To discuss if this way of conflicting build_requires is actually useful or not
-            downstream_require = Requirement(require.ref, headers=False, libs=False, build=True,
-                                             run=False, visible=self.visible, direct=False)
-            return downstream_require
-
-        if self.build:  # Build-requires
-            # If the above is shared or the requirement is explicit run=True
-            if dep_pkg_type is PackageType.SHARED or require.run:
-                downstream_require = Requirement(require.ref, headers=False, libs=False, build=True,
-                                                 run=True, visible=False, direct=False)
-                return downstream_require
-            return
+        if self.build and dep_pkg_type is not PackageType.SHARED and not require.run:  # Build-requires
+            return # don't propagate non-run packages
```

Comment:
Propagation of ``visible=True`` is not only about traits, packages do not only conflict because having ``headers``, even if they don't 2 different packages (for the same package name) with ``visible=True`` will conflict.

If they don't conflict, there could be 2 different version of the same ``myscripts/version`` package being propagated down the graph as visible, but not conflicting would be a problem.

Reply:
Hmm okay but shouldn't then requirements always be propagated when they are `visible`? This is definitely not the case in the current implementation.